src/main/java/me/zeroeightsix/kami/feature/module/player/Blink.java

```diff
@@ -62,9 +67,14 @@ public void onEnable() {
 
     @Override
     public void onDisable() {
-        if (packets.size() > maxPacketAmount && packetDiscard) {
-            if (clonedPlayer != null) // We don't want a NPE when a player tries to disable blink after logging in
-                mc.player.setPos(clonedPlayer.getPos().x, clonedPlayer.getPos().y, clonedPlayer.getPos().z); // Snap back to where we were
+        if (packets.size() > maxPacketAmount &&
+                isCancelable && clonedPlayer != null &&
+                clonedPlayer.getPos().distanceTo(mc.player.getPos()) > 10) { // If its more than 10 blocks
```

Comment:
Why 10 blocks? If the max packet threshold isn't working, I'd be more suspicious of the packet event not actually cancelling the sent packet. Adding a 10 block limit seems kind of arbitrary and well, useless?

Reply:
From my testing if you snap back at a distance lesser then that of ten blocks, the next movement input you do, the game will attempt to teleport you to where you stopped blinking. This teleport works on servers with no anticheat (thereby making the ability to cancel do nothing at that distance) but on servers with anticheat it gets detected for a speedhack.

Reply:
In that case you should try to figure out why the game is trying to teleport you back to where you stopped blinking.

Instead of `setPos`, can you try using `updatePostion`?